lib/rest-api.php

```diff
@@ -219,6 +219,7 @@ function gutenberg_add_permalink_template_to_posts( $response, $post, $request )
 
 	$response->data['permalink_template'] = $sample_permalink[0];
 	$response->data['generated_slug']     = $sample_permalink[1];
+	$response->data['is_front_page']      = 'page' === get_option( 'show_on_front' ) && (int) get_option( 'page_on_front' ) === $post->ID;
```

Comment:
It looks like this function is now deprecated so this change should be applied directly in core:
https://github.com/WordPress/gutenberg/blob/master/lib/rest-api.php#L102